Detectors/GlobalTracking/src/MatchHMP.cxx

```diff
@@ -258,7 +259,24 @@ void MatchHMP::addConstrainedSeed(o2::track::TrackParCov& trc, o2::dataformats::
     mTracksLblWork[o2::globaltracking::MatchHMP::trackType::CONSTR].emplace_back(mRecoCont->getTPCITSTrackMCLabel(srcGID));
   }
 
-  mTracksIndexCache[o2::globaltracking::MatchHMP::trackType::CONSTR].push_back(it);
+  auto prop = o2::base::Propagator::Instance();
+  float bxyz[3];
+  prop->getFieldXYZ(trc.getXYZGlo(), bxyz);
+  double bz = -bxyz[2];
+
+  float pCut = 0.;
+
+  if (TMath::Abs(bz - 5.0) < 0.5)
+    pCut = 0.450;
+  if (TMath::Abs(bz - 2.0) < 0.5)
+    pCut = 0.200;
+
+  if (trc.getP() > pCut &&
+      TMath::Abs(trc.getEta()) < 0.52 &&
+      (TMath::RadToDeg() * trc.getPhi() < 90. || TMath::RadToDeg() * trc.getPhi() > 270.)) {
```

Comment:
please use 
```
if (trc.getP() > pCut && TMath::Abs(trc.getTgl()) < 0.544 && TMath::Abs(trc.getPhi() -TMath::Pi())<(TMath::Pi()*0.5)) { ...
```
to avoid unnecessary calculations.

Reply:
ok, I will do it, but what getTgl() is? Why do you not like getEta()?

Reply:
getTgl is a getter for the tan(lambda) (or 1/tan(theta)) : on of the track parameters, so it comes for no cost.
getEta() is calculated from it via quite expensive calls: https://github.com/AliceO2Group/AliceO2/blob/5b4a79edcad26f04d1052ba8093155e327c5c3fc/DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackParametrization.h#L597-L608